examples/rust/get_started/examples/11-attribute-based-authentication.rs

```diff
@@ -0,0 +1,91 @@
+///  - the control node in `11-attribute-based-authentication-control-plane.rs`
+///  - the edge node in `11-attribute-based-authentication-edge-plane.rs`
+///
+/// This will set up a TCP outlet on the control node, connected to the Python webserver
+/// and a TCP inlet on the edge node which can be used to send HTTP requests (at 127.0.0.1:7000).
+///
+/// Then if you execute `curl --fail --head --max-time 10 127.0.0.1:7000` you should get back
+/// a successful response like:
+///
+/// HTTP/1.0 200 OK
+/// Server: SimpleHTTP/0.6 Python/3.9.6
+/// Date: Tue, 07 Feb 2023 15:05:59 GMT
+/// Content-type: text/html; charset=utf-8
+/// Content-Length: 870
+///
+/// and observe that a successful connection has been made on the Python webserver:
+///
+/// ± python3 -m http.server --bind 127.0.0.1 5000                                                                                                                                                                                                                                                                                                                                                                                      default
+///  Serving HTTP on 127.0.0.1 port 5000 (http://127.0.0.1:5000/) ...
+///  127.0.0.1 - - [06/Feb/2023 15:52:20] "HEAD / HTTP/1.1" 200 -
+///
+/// TOPOLOGY
+///
+/// The network we establish between the control node, the edge node and the Orchestrator is the following
+///
+///                   +-------------------------+
+///                   | Edge node               | Inlet <-- 127.0.0.1:7000
+///                   |                         | connected to "outlet" via secure channel
+///                   +-------------------------+
+///                     |                 |
+///  secure channel to  |                 | create secure channel to control
+///  project            |                 | via the forwarder
+///                     v                 |
+///                   +-------------------+-------+                 +--------------+
+///                   | Orchestrator      |       | get credentials | Authority    |
+///                   |                   |       |---------------->|              |
+///                   |                   |       |                 |              |
+///                   +------------forwarder------+                 +--------------+
+///                     ^          to control
+///                     |         ^       |
+///  secure channel to  |  create |       |
+///  project            |         |       v
+///                     |         |    "untrusted"
+///                   +--------------secure channel listener--+
+///                   |                                       |
+///                   | Control node                          | "outlet" --> 127.0.0.1:5000
+///                   |                                       |
+///                   +---------------------------------------+
```

Comment:
This looks good!.
One comment,   there is one role missing here, the `enroller`  .    get_credentials  can be used _after_   that identity is a member,  and that can be done only by the `enroller`  (either  by using a one-time-token or directly adding the identity identifier of the new member).    

In the example,  looks like  *both*  the edge and control node are able to act as  `enroller`  due to the underling identity on the _machine_ they run.

I find that a bit confusing and hiding how and _why_  things works. 
Would it be better if instead it's done as:

```
$ ockam enroll
$ ockam project information --output json  > project.json
$ ockam project enroll --attribute  component=edge
     -> gives EDGE_ENROLLMENT_TOKEN
$ ockam project enroll --attribute  component=control_plane
     > gives CONTROL_PLANE_ENROLLMENT_TOKEN
```

Now,  from any machine,  you can start:    
```
 $ attribute-based-authentication-control-plane   project.json   $CONTROL_PLANE_ENROLLMENT_TOKEN
 $ attribute-based-authentication-edge   project.json   $EDGE_ENROLLMENT_TOKEN
 ```

if we want to make it simple to run locally,    a bash script can be used   (that do the enroll,  generate the tokens,   and start the components passing the right argument).  That bash script will server as doc of how things are done in general

Reply:
Thanks @polvorin for your comment. It is indeed important to show that the enrollment is a separate phase of the process. I tried a different approach to show this though in order to stay inside the development environment as much as possible. I separated the code between a setup phase where we find out what the token and the project information should be and a `start_node` phase where we start the node with this information.

I did this because I found very useful to be able to stay in the IDE while developing this scenario, and even be able to access the debugger to understand what was wrong. So I think that this format will also help people iterate faster when they try to expand on this use case.